CRM/Utils/System/Drupal.php

```diff
@@ -72,6 +72,23 @@ public function createUser(&$params, $mail) {
     return $form_state['user']->uid;
   }
 
+  /**
+   * Appends a Drupal 7 Javascript file when the CRM Menubar Javascript file has
+   * been included.
+   */
+  public function appendCoreResources(\Civi\Core\Event\GenericHookEvent $event) {
+    $crmDrupal7JsFilePath = 'js/crm.drupal7.js';
+    $smartMenuFilePath = 'js/crm.menubar.js';
+    $smartMenuFileIndex = array_search($smartMenuFilePath, $event->list);
+    $hasNotIncludedTheCrmMenu = $smartMenuFileIndex === FALSE;
+
+    if ($hasNotIncludedTheCrmMenu) {
+      return;
+    }
+
+    array_splice($event->list, $smartMenuFileIndex + 1, 0, [$crmDrupal7JsFilePath]);
```

Comment:
Adding this file *after* `crm.menubar.js` seems like it could lead to a race condition. The way you've got it structured here:
1.  `crm.menubar.js` loads first and fires the `crmLoad` event.
2. `rm.drupal7.js` loads second and *listens* for the `crmLoad` event.

See the problem?

I would bump up the order of this file, and then in crm.drupal7.js  you can just take out the import of `CRM.menubar`. It's only a slight convenience to alias it as `crmMenubar` but should be perfectly fine to just access it directly in the callback. The listener will be set up before that variable is defined, but once the event fires it *will* be defined so you're good.